components/automate-ui/src/app/pages/policy/add-members/policy-add-members.component.ts

```diff
@@ -305,4 +332,74 @@ export class PolicyAddMembersComponent implements OnInit, OnDestroy {
     this.addOrRemoveQueuedMember(true, member);
     this.closeModal();
   }
+
+  public showInputs(inputName: string): boolean {
+    const typeValue = this.expressionForm.get('type').value;
+    const identityValue = this.expressionForm.get('identity').value;
+
+    switch (inputName) {
+      case 'identity':
+        if (typeValue === 'USER' || typeValue === 'TEAM') {
+            this.allIdentity = typeValue;
+            this.updateValidations('identity', true);
+            return true;
+          } else {
+            this.updateValidations('identity', false);
+            this.expressionForm.get('identity').reset();
+            return false;
+          }
+        break;
+      case 'name':
+        if ( typeValue === 'TOKEN' || (identityValue && identityValue !== '*') ) {
```

Comment:
if the user has chosen Toke as their type and the identity value is not a `*`, we want to show the name element.